src/memex/models/document.py

```diff
@@ -393,29 +395,35 @@ def merge_documents(session, documents, updated=None):
             m.document = master
             m.updated = updated
 
-        session.delete(doc)
-
     try:
         session.flush()
+        session.query(Annotation) \
+            .filter(Annotation.document_id.in_(duplicate_ids)) \
+            .update({Annotation.document_id: master.id}, synchronize_session='fetch')
```

Comment:
Should there be a test that `merge_documents()` updates `Annotation.document_id`s? (I don't think I saw one)